src/Dock.Avalonia.Themes.Default/PinHostControl.xaml

```diff
@@ -0,0 +1,107 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:id="clr-namespace:Dock.Avalonia;assembly=Dock.Avalonia"
+        xmlns:idc="clr-namespace:Dock.Avalonia.Controls;assembly=Dock.Avalonia"
+        xmlns:converters="clr-namespace:Dock.Avalonia.Converters;assembly=Dock.Avalonia"
+        xmlns:Dock="clr-namespace:Dock.Model;assembly=Dock.Model"
+        xmlns:Controls="clr-namespace:Dock.Model.Controls;assembly=Dock.Model">
+
+    <Style Selector="idc|PinHostControl">
+        <Style.Resources>
+            <converters:PinDockToGridLengthConverter x:Key="PinDockToGridLengthConverter"/>
+        </Style.Resources>
+        <Setter Property="Template">
+            <ControlTemplate>
+                <DockPanel Background="Transparent">
+                    <ContentControl Content="{Binding Left}" DockPanel.Dock="Left"/>
+                    <ContentControl Content="{Binding Right}" DockPanel.Dock="Right"/>
+                    <Grid>
+                        <Panel x:Name="LeftPanel"
```

Comment:
These panels host expanding content of pin dock. I've reused DockToolChrome, but probably it needs custom control with pin flipping to horizontal when dock is on auto-hide.